### PR TITLE
Exception handling enhancement

### PR DESCRIPTION
**Pull request by sbernard31**
Improve message error of Leshan exception and clean code using String.format like constructor for exception.

### CODE REVIEW

File: leshan-core/src/main/java/org/eclipse/leshan/core/request/BootstrapWriteRequest.java [hunk 1]
```diff
@@ -36,43 +36,46 @@ public BootstrapWriteRequest(LwM2mPath target, LwM2mNode node, ContentFormat for
...
         if (node == null)
-            throw new InvalidRequestException("new node value is mandatory");
+            throw new InvalidRequestException("new node value is mandatory for %s", target);
 
         // Validate node and path coherence
         if (getPath().isResource()) {
             if (!(node instanceof LwM2mResource)) {
-                throw new InvalidRequestException(String.format("path '%s' and node type '%s' does not match",
-                        target.toString(), node.getClass().getSimpleName()));
+                throw new InvalidRequestException("path '%s' and node type '%s' does not match", target,
```
Comment (sophokles73) [inline leshan-core/src/main/java/org/eclipse/leshan/core/request/BootstrapWriteRequest.java]: should be `do not match`

File: leshan-core/src/main/java/org/eclipse/leshan/core/request/BootstrapWriteRequest.java [hunk 2]
```diff
@@ -36,43 +36,46 @@ public BootstrapWriteRequest(LwM2mPath target, LwM2mNode node, ContentFormat for
...
+                        node.getClass().getSimpleName());
             }
         } else if (getPath().isObjectInstance()) {
             if (!(node instanceof LwM2mObjectInstance)) {
-                throw new InvalidRequestException(String.format("path '%s' and node type '%s' does not match",
-                        target.toString(), node.getClass().getSimpleName()));
+                throw new InvalidRequestException("path '%s' and node type '%s' does not match", target,
```
Comment (sophokles73) [inline leshan-core/src/main/java/org/eclipse/leshan/core/request/BootstrapWriteRequest.java]: should be `do not match`

File: leshan-core/src/main/java/org/eclipse/leshan/core/request/BootstrapWriteRequest.java [hunk 3]
```diff
@@ -36,43 +36,46 @@ public BootstrapWriteRequest(LwM2mPath target, LwM2mNode node, ContentFormat for
...
+                        node.getClass().getSimpleName());
             }
         } else if (getPath().isObject()) {
             if (!(node instanceof LwM2mObject)) {
-                throw new InvalidRequestException(String.format("path '%s' and node type '%s' does not match",
-                        target.toString(), node.getClass().getSimpleName()));
+                throw new InvalidRequestException("path '%s' and node type '%s' does not match", target,
```
Comment (sophokles73) [inline leshan-core/src/main/java/org/eclipse/leshan/core/request/BootstrapWriteRequest.java]: should be `do not match`

File: leshan-core/src/main/java/org/eclipse/leshan/core/request/DeleteRequest.java [hunk 4]
```diff
@@ -47,7 +47,7 @@ public DeleteRequest(String path) throws InvalidRequestException {
     private DeleteRequest(LwM2mPath target) {
         super(target);
         if (!target.isObjectInstance())
-            throw new InvalidRequestException("Only object instance can be delete.");
+            throw new InvalidRequestException("Invalid path %s : Only object instance can be delete.", target);
```
Comment (sophokles73) [inline leshan-core/src/main/java/org/eclipse/leshan/core/request/DeleteRequest.java]: should be `...object instances can be deleted"

File: leshan-core/src/main/java/org/eclipse/leshan/core/request/WriteRequest.java [hunk 5]
```diff
@@ -288,54 +288,57 @@ public WriteRequest(int objectId, int objectInstanceId, int resourceId, Map<Inte
...
-            throw new InvalidRequestException(
-                    String.format("Invalid mode for '%s': update is not allowed on resource", target.toString()));
+            throw new InvalidRequestException("Invalid mode for '%s': update is not allowed on resource", target);
 
         // Validate node and path coherence
         if (getPath().isResource()) {
             if (!(node instanceof LwM2mResource)) {
-                throw new InvalidRequestException(String.format("path '%s' and node type '%s' does not match",
-                        target.toString(), node.getClass().getSimpleName()));
+                throw new InvalidRequestException("path '%s' and node type '%s' does not match", target,
```
Comment (sophokles73) [inline leshan-core/src/main/java/org/eclipse/leshan/core/request/WriteRequest.java]: `do not match`

File: leshan-core/src/main/java/org/eclipse/leshan/core/request/WriteRequest.java [hunk 6]
```diff
@@ -288,54 +288,57 @@ public WriteRequest(int objectId, int objectInstanceId, int resourceId, Map<Inte
...
+                        node.getClass().getSimpleName());
             }
         } else if (getPath().isObjectInstance()) {
             if (!(node instanceof LwM2mObjectInstance)) {
-                throw new InvalidRequestException(String.format("path '%s' and node type '%s' does not match",
-                        target.toString(), node.getClass().getSimpleName()));
+                throw new InvalidRequestException("path '%s' and node type '%s' does not match", target,
```
Comment (sophokles73) [inline leshan-core/src/main/java/org/eclipse/leshan/core/request/WriteRequest.java]: `do not match`

**Comment by sbernard31** [comment]
Thx @sophokles73, I will fix that :)

**Comment by sbernard31** [comment]
@sophokles73  Done.

**Comment by sbernard31** [comment]
Integrated in master (commit 744150b5e36e3468a64a0292bdc0cb196770a177, c01624ae71e121f45f1ff0dc9650273503a148a5, ef6772193281b955f07afdf584ce0a611f9bec71)

**Pull request closed without merging**